src/ToDo.UI/Controls/NavigationViewRequestHandler.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Uno.Extensions.Navigation.UI;
+
+namespace ToDo.Views;
+
+public class NavigationViewRequestHandler : ControlRequestHandlerBase<NavigationView>
+{
+	public override IRequestBinding? Bind(FrameworkElement view)
+	{
+		var viewToBind = view;
+		var viewList = view as NavigationView;
+		if (viewList is null)
+		{
+			return default;
+		}
+
+		Func<FrameworkElement, object, Task> action = async (sender, data) =>
+		{
+			var navdata = sender.GetData() ?? data;
```

Comment:
I don't really know such deep details about navigation, but I think this line can lead to confusing behavior where explicit data could be discarded when also provided by the sender... no?

Reply:
Yes this is by design - default behavior (ie "data") is the item being tapped on. However, this can be overridden by setting the Navigation.Data attached property (ie sender.GetData())